packages/block-library/src/legacy-widget/edit/form.js

```diff
@@ -36,127 +20,37 @@ export default function Form( { id, idBase, instance, setInstance } ) {
 		<Control
 			id={ id }
 			idBase={ idBase }
-			html={ html }
+			content={ content }
 			onChange={ setFormDataDebounced }
 			onSave={ setFormData }
+			// Force a remount when the widget's form HTML changes. This clears
+			// out any mutations to the DOM that widget scripts have made.
+			key={ content.key }
```

Comment:
I think using `key` to force a remount is fine, but I don't like using an arbitrary `key` variable for it. Could we just use `html` as the key? It might impact performance, but maybe it's insignificant.

Reply:
We can't because then the _Cancel_ flow won't work. We only want to remount the component when the form HTML changes _due to something other than user input_.

To illustrate:

1. User has a text widget. The form HTML (`content.html`) is `"<textarea>Foo</textarea>"`.
2. User appends "Bar". `useForm` detects change to `formData`, sends XHR request. It updates `pendingInstance` to `{ "content": "FooBar" }` but does **not** update form HTML (`content.html`) because we do not want to cause a bad UX by remounting the form after user input.
3. User presses _Cancel_. `pendingInstance` is reset to `instance` which is `{ "content": "Foo" }`. `useForm` detects this, sends XHR request, and sets form HTML (`content.html`) back to `"<textarea>Foo</textarea>"`.
4. Form component would **not** remount because the key has not changed (`"<textarea>Foo</textarea>" === "<textarea>Foo</textarea>"`).

Happy to change this if you can think of a better approach. An explicit `key` was the best I could come up with 😅. Maybe it should be called `content.version`.

Reply:
Do we still have a cancel flow now? 😅

Reply:
Right you are! Fixed in ef778996f0c4b032a6a3aaa09ffd76389dd18659.